src/main/java/org/example/daos/FileImportDao.java

```diff
@@ -0,0 +1,228 @@
+package org.example.daos;
+import javax.xml.transform.Result;
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import org.example.enums.Location;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.sql.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class FileImportDao {
+
+    public int importRoles(final List<String> listOfLines) throws SQLException {
+        try (Connection connection = DatabaseConnector.getConnection()) {
+
+            String insertStatement = "INSERT INTO Job_Roles"
+                    + "(roleName, description, responsibilities, "
+                    + "linkToJobSpec, capability, band, closingDate, `status`, "
+                    + "positionsAvailable) "
+                    + "VALUES "
+                    + "(?,?,?,?,?,?,?,?,?);";
+
+            String insertLocationStatement = "INSERT INTO Job_Location_Connector"
+                    + "(roleId, roleLocationId)"
+                    + "VALUES"
+                    + "(?, ?);";
+
+
+            PreparedStatement statement = connection.prepareStatement(
+                    insertStatement,
+                    PreparedStatement.RETURN_GENERATED_KEYS);
+            PreparedStatement locationStatement = connection.prepareStatement(insertLocationStatement);
+
+            System.out.println("Number of lines " + listOfLines.size());
+
+            //This stuff goes into the service
+
+            Map<String, Object>[] arrayOfMaps = new HashMap[listOfLines.size()];
+
+            for (int i = 0; i < arrayOfMaps.length; i++) {
+                arrayOfMaps[i] = new HashMap<>();
+            }
+
+            //Everything after this stays in the Dao
+
+            SimpleDateFormat sqlDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+            String[] lineContents;
+
+            for (String line : listOfLines) {
+                lineContents = line.split(",");
```

Comment:
This should be converted into an object earlier on in the process so you don't have to do all of this line parsing